File: PCMania/src/Cliente/Cliente.java
package Cliente;

import Computador.Computador;

public class Cliente {
    private String nome;
    private String cpf;

    private Computador[] computadores;


    public Cliente(String nome, String cpf){
        this.nome = nome;
        this.cpf = cpf;
    }

    public float calculaTotalCompra(){
        float precoTotal = 0;
        for (int i = 0; i < computadores.length; i++) {
            if(computadores[i] == null){
                continue;
            } 
            precoTotal += computadores[i].getPreco();

        }
        return precoTotal;
    }

    //Setter
    public void setComputadores(Computador[] computadores) {
        this.computadores = computadores;
    }

    //Getters
    public String getCpf() {
        return cpf;
    }

    public String getNome() {
        return nome;
    }
}

File: POO_CPP/PCManiaCPP/src/Cliente/Cliente.cpp
#include "Cliente.h"

Cliente::Cliente(string nome, string cpf){
    this->nome = nome;
    this->cpf = cpf;
    // O vector 'computadores' é inicializado vazio por padrão
}

float Cliente::calculaTotalCompra(){
    float precoTotal = 0;
    for (size_t i=0; i<computadores.size() ; i++){
        // Acessamos o elemento na posição 'i' com .at(i)
        precoTotal += computadores.at(i).getPreco();
    }
    return precoTotal;

}

void Cliente::setComputadores(vector<Computador> computadores){
    this->computadores = computadores;
}

string Cliente::getCpf(){
    return this->cpf;
}

string Cliente::getNome(){
    return this->nome;
}